grafana-plugin/src/pages/integration/OutgoingTab/ConnectIntegrationModal.tsx

```diff
@@ -13,13 +13,15 @@ import { useStore } from 'state/useStore';
 import { useCommonStyles, useIsLoading } from 'utils/hooks';
 
 import ConnectedIntegrationsTable from './ConnectedIntegrationsTable';
+import { useCurrentIntegration } from './OutgoingTab.hooks';
 import { getStyles } from './OutgoingTab.styles';
 
 const DEBOUNCE_MS = 500;
 
 export const ConnectIntegrationModal = observer(({ onDismiss }: { onDismiss: () => void }) => {
-  const { alertReceiveChannelStore } = useStore();
-  const isLoading = useIsLoading(ActionKey.FETCH_INTEGRATIONS);
+  const { alertReceiveChannelStore, alertReceiveChannelConnectedChannelsStore } = useStore();
+  const currentIntegration = useCurrentIntegration();
```

Comment:
`currentIntegration` is used here only when user tries to connect integration within modal. Connecting integrations is possible only when user selects them in the table. For the table we have a loading state so only when data is present, user can select them and then connect them. Also modal is not opened automatically so it takes some secs / ms for the user to open it first.

Second case where we use this hook is to show snow url. In the worst case it will be blank for the first couple of ms and then filled out with the data that comes from current integration. We use optional chaining there so no risk to get Cannot read X of undefined